src/network/tc/networkd-tc-core.c

```diff
@@ -0,0 +1,35 @@
+/* SPDX-License-Identifier: LGPL-2.1+
+ * Copyright © 2019 VMware, Inc.
+ */
+
+#include "alloc-util.h"
+#include "fileio.h"
+#include "fd-util.h"
+#include "networkd-tc-core.h"
+#include "time-util.h"
+
+static double ticks_in_usec = 1;
+static double clock_factor = 1;
+
+int tc_init(void) {
```

Comment:
Instead of calling this function in `manager_new()`, how about caching `ticks_in_usec` in the function? I mean
```
double get_ticks_in_usec(void) {
    static ticks_in_usec = -1;

    if (ticks_in_usec > 0)
        return ticks_in_usec;

    /* read /proc/net/psched and calculate ticks_in_usec. */
}
```

Reply:
Ye nice idea